boot/zephyr/main.c

```diff
@@ -376,12 +426,50 @@ void main(void)
     __ASSERT(rc >= 0, "Error of the reading the detect pin.\n");
     if (detect_value == CONFIG_BOOT_SERIAL_DETECT_PIN_VAL &&
         !boot_skip_serial_recovery()) {
-        BOOT_LOG_INF("Enter the serial recovery mode");
-        rc = boot_console_init();
-        __ASSERT(rc == 0, "Error initializing boot console.\n");
-        boot_serial_start(&boot_funcs);
-        __ASSERT(0, "Bootloader serial process was terminated unexpectedly.\n");
-    }
+            
+        k_sleep(K_MSEC(50));
```

Comment:
this is part of debounce time as well.

Reply:
Only used now when `CONFIG_BOOT_SERIAL_DETECT_DELAY` is defined > 0. Ensures there's no bouncing for that first 50 ms.